source/extensions/filters/http/sxg/filter.cc

```diff
@@ -0,0 +1,497 @@
+#include "source/extensions/filters/http/sxg/filter.h"
+
+#include <cstdlib>
+#include <regex>
+#include <string>
+#include <chrono>
+
+#include <openssl/evp.h>
+#include <openssl/bio.h>
+#include <openssl/pem.h>
+#include <openssl/x509.h>
+
+#include "envoy/server/filter_config.h"
+// #include "pinterest/common/knox/client.h"
+
+#include "absl/strings/escaping.h"
+#include "envoy/stats/scope.h"
+#include "envoy/http/codes.h"
+#include "source/common/common/utility.h"
+#include "source/common/http/headers.h"
+#include "source/common/stats/utility.h"
+
+#include "libsxg.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace SXG {
+
+Http::RegisterCustomInlineHeader<Http::CustomInlineHeaderRegistry::Type::RequestHeaders>
+    accept_handle(Http::CustomHeaders::get().Accept);
+
+template <class T>
+const std::vector<std::string> initializeHeaderPrefixFilters(const T& filters_proto) {
```

Comment:
@rgs1 is this the best way to initialize a list of strings from the proto buffer? this replaces the `x-pinterest-` header filter.